front/pages/w/[wId]/ds/[name]/upsert.tsx

```diff
@@ -205,8 +206,11 @@ export default function DataSourceUpsert({
                         id="document"
                         readOnly={readOnly}
                         className={classNames(
-                          "block w-full min-w-0 flex-1 rounded-md text-sm",
-                          "border-gray-300 focus:border-violet-500 focus:ring-violet-500"
+                          "block w-full min-w-0 flex-1 rounded-md border-gray-300 text-sm",
+                          "border-gray-300",
+                          readOnly
```

Comment:
cc @spolu , kind of unrelated, but for "Settings" and "Upsert", I removed the "focus" indicators on the inputs when they are readonly (felt a bit weird IMO to be able to focus them but not write)

Reply:
yep :+1: